packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -1,33 +1,249 @@
 /*
- * Copyright (c) 2017, salesforce.com, inc.
- * All rights reserved.
- * Licensed under the BSD 3-Clause license.
- * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
- */
+* Copyright (c) 2017, salesforce.com, inc.
+* All rights reserved.
+* Licensed under the BSD 3-Clause license.
+* For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+*/
 import * as fs from 'fs';
+import * as mkdirp from 'mkdirp';
 import * as path from 'path';
-import { SObjectCategory, SObjectDescribe } from '../describe';
+import {
+  ChildRelationship,
+  Field,
+  SObject,
+  SObjectCategory,
+  SObjectDescribe
+} from '../describe';
 
 export class FauxClassGenerator {
-  public async generate(projectPath: string, type: SObjectCategory) {
+  private SFDX_DIR = '.sfdx';
+  private TOOLS_DIR = 'tools';
+  private SOBJECTS_DIR = 'sobjects';
+
+  private createIfNeededOutputFolder(folderPath: string): boolean {
+    if (!fs.existsSync(folderPath)) {
+      mkdirp.sync(folderPath);
+      return fs.existsSync(folderPath);
+    }
+    return true;
+  }
+
+  public async generate(
```

Comment:
This method seems too long. Please refactor.

Reply:
It seems like most of it is long because of all the logging? Is the logging required? Does it provide us with useful information? Or is it just generating a bunch of noise for the user?

Reply:
And I do agree that the method is too long  - will fix that ASAP...   I think that the logging is actually useful information. Except probably the one on the output folders - I'll get rid of that.